cli/src/commands/git/fetch.rs

```diff
@@ -72,13 +83,50 @@ pub fn cmd_git_fetch(
     args: &GitFetchArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let remotes = if args.all_remotes {
-        git::get_all_remote_names(workspace_command.repo().store())?
+    let remote_patterns = if args.all_remotes {
+        vec![StringPattern::everything()]
     } else if args.remotes.is_empty() {
         get_default_fetch_remotes(ui, &workspace_command)?
     } else {
         args.remotes.clone()
     };
+
+    let all_remotes = git::get_all_remote_names(workspace_command.repo().store())?;
+
+    let (unmatched_patterns, remotes): (Vec<_>, Vec<_>) =
+        remote_patterns.into_iter().partition_map(|pattern| {
+            let matching_remotes = all_remotes
+                .iter()
+                .filter(|r| pattern.matches(r))
+                .collect::<Vec<_>>();
+            if matching_remotes.is_empty() {
+                Either::Left(pattern)
+            } else {
+                Either::Right(matching_remotes)
+            }
+        });
```

Comment:
I think it's simpler to use plain `for` loop instead of building nested `remotes: Vec<Vec<&String>>`. Maybe we can also use `HashSet` or `IndexSet` for deduplication, but I'm not sure if that makes the code more readable. The performance wouldn't matter here.
https://docs.rs/indexmap/2.7.1/indexmap/set/struct.IndexSet.html